freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UmsUsersStateProvider.java

```diff
@@ -129,95 +133,38 @@ private WorkloadCredential getCredentials(String userCrn, Optional<String> reque
-            }
-        }
-
-        return false;
-    }
-
     @SuppressWarnings("ParameterNumber")
     private void handleUser(UmsUsersState.Builder umsUsersStateBuilder, UsersState.Builder usersStateBuilder, Map<String, FmsGroup> crnToFmsGroup,
-                            String actorCrn, String memberCrn, FmsUser fmsUser, String environmentCrn, Optional<String> requestId) {
-        try {
-            GetRightsResponse rightsResponse = grpcUmsClient.getRightsForUser(actorCrn, memberCrn, environmentCrn, requestId);
-            if (isEnvironmentUser(environmentCrn, rightsResponse)) {
-                usersStateBuilder.addUser(fmsUser);
-                rightsResponse.getGroupCrnList().stream().forEach(gcrn -> {
-                    FmsGroup group = crnToFmsGroup.get(gcrn);
-                    // If the group is null, then there has been a group membership change after we started the sync
-                    // the group and group membership will be updated on the next sync
-                    if (group != null) {
-                        usersStateBuilder.addMemberToGroup(group.getName(), fmsUser.getName());
-                    }
-                });
-
-                // Since this user is eligible, add this user to internal group
-                usersStateBuilder.addMemberToGroup(UserServiceConstants.CDP_USERSYNC_INTERNAL_GROUP, fmsUser.getName());
+                            String memberCrn, FmsUser fmsUser, EnvironmentAccessRights environmentAccessRights, Optional<String> requestId) {
+        if (environmentAccessRights.hasEnvironmentAccessRight()) {
+            usersStateBuilder.addUser(fmsUser);
+            String username = fmsUser.getName();
+            String accountId = Crn.safeFromString(memberCrn).getAccountId();
+            grpcUmsClient.listGroupsForMember(IAM_INTERNAL_ACTOR_CRN, accountId, memberCrn, requestId).forEach(gcrn -> {
+                FmsGroup group = crnToFmsGroup.get(gcrn);
+                // If the group is null, then there has been a group membership change after we started the sync
+                // the group and group membership will be updated on the next sync
+                if (group != null) {
+                    usersStateBuilder.addMemberToGroup(group.getName(), username);
+                }
+            });
 
-                List<String> workloadAdministrationGroupNames = rightsResponse.getWorkloadAdministrationGroupNameList();
-                LOGGER.debug("workloadAdministrationGroupNameList = {}", workloadAdministrationGroupNames);
-                workloadAdministrationGroupNames.forEach(groupName -> {
-                    usersStateBuilder.addGroup(nameToGroup(groupName));
-                    usersStateBuilder.addMemberToGroup(groupName, fmsUser.getName());
-                });
+            // Since this user is eligible, add this user to internal group
+            usersStateBuilder.addMemberToGroup(UserServiceConstants.CDP_USERSYNC_INTERNAL_GROUP, username);
```

Comment:
could you move this to a method called sth like `addEligibleUserToInnternalGroup` so we can drop  the comment

Reply:
moved